### ISSUE TITLE
Wrong type-checking on merging interface method and non-method members

### ISSUE THREAD

**Issue by thorn0** [Microsoft/TypeScript#15203]
**TypeScript Version:**  2.2.1, 2.3.0

**Code**

```ts
interface Foo {
    bar(): number;
}

interface Foo {
    bar: () => number; // error here
}

interface Foo2 {
    bar(): number;
}

interface Foo2 {
    bar: number; // no error here
}
```

**Expected behavior:**
- `Foo` should compile.
- `Foo2` shouldn't compile.

**Actual behavior:**
- `Foo` doesn't compile with an erroneous error message: `Subsequent variable declarations must have the same type.  Variable 'bar' must be of type 'number', but here has type '() => number'.`
- `Foo2` compiles.

**Comment by thorn0** [comment]
Somehow it has been fixed. [playground](http://www.typescriptlang.org/play/#src=interface%20Foo%20%7B%0D%0A%20%20%20%20bar()%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20Foo%20%7B%0D%0A%20%20%20%20bar%3A%20()%20%3D%3E%20number%3B%20%2F%2F%20no%20error%20here%20now%0D%0A%7D%0D%0A%0D%0A%0D%0Ainterface%20Foo2%20%7B%0D%0A%20%20%20%20bar()%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20Foo2%20%7B%0D%0A%20%20%20%20bar%3A%20number%3B%20%2F%2F%20error%20here%20now%0D%0A%7D)
